### PR TITLE
[s] more immersive slipping experience

### PR DESCRIPTION
**Pull request by Incoming5643**
In the real world a slip can range wildly from a minor scuffle to a potentially fatal event. At long last our spacemen can have the same experience of cruel probabilistic pain.

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -226,9 +226,8 @@
 		playsound(src, "bodyfall", 50, 1)
 
 /turf/handle_slip(mob/living/carbon/C, s_amount, w_amount, obj/O, lube)
-	if(has_gravity(src))
+	if(has_gravity(src) && C.lying)
```
Comment (Incoming5643) [inline code/game/turfs/turf.dm]: else the clown PDA will kill everyone

Comment (Bawhoppen) [inline code/game/turfs/turf.dm]: you say that like its a bad thing

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -241,12 +240,63 @@
...
+		if(C.dna)
+			if(C.dna.features["tail_human"] == "Cat" && C.dna.features["ears"] == "Cat")
+				C.SpinAnimation(7,1)
+				visible_message("<span class='notice'>[C.name] slips... but manages to stick the landing with their cat like reflexs!</span>")
+				return 0
+		var/list/impact_point = pick("Head", "Chest", "Back", "Ass", "Hands", "Groin")
+		var/message
+		switch(impact_point)
+			if("Head") //oh boy
+				message = "[C.name] slips backwards straight onto their head! You hear a"
```
Comment (PKPenguin321) [inline code/game/turfs/turf.dm]: what is the purpose of this line

**Comment by PKPenguin321** [comment]
i've considered adding something like this to the actual game for skateboard crashing

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -241,12 +240,63 @@
...
+			if("Back")
+				message = "[C.name] slips backwards flat on their back!"
+				C.apply_damage(10, BRUTE, "chest")
+
+			if("Ass") //Tactically speaking, this is the best place to land
+				message = "[C.name] slips, landing squarely on their ass!"
+				C.apply_damage(5, BRUTE, "chest") //Where's my ass damage zone?
+
+			if("Hands")
+				message = "[C.name] slips forwards, holding on their hands to catch their fall!"
```
Comment (ChangelingRain) [inline code/game/turfs/turf.dm]: holding out

File: code/game/turfs/turf.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/turfs/turf.dm]: this prevents 260-270 from running

Comment (Incoming5643) [inline code/game/turfs/turf.dm]: just a remnant, shouldn't ever be seen in game but will remove if i have to change anything else.

File: code/game/turfs/turf.dm [hunk 4]
```diff
@@ -241,12 +240,62 @@
...
+				message = "[C.name] slips fowards flat onto their chest, knocking the air out of their lungs!"
+				C.setOxyLoss(50)
+
+			if("Back")
+				message = "[C.name] slips backwards flat on their back!"
+				C.apply_damage(10, BRUTE, "chest")
+
+			if("Ass") //Tactically speaking, this is the best place to land
+				message = "[C.name] slips, landing squarely on their ass!"
+				C.apply_damage(5, BRUTE, "chest") //Where's my ass damage zone?
```
Comment (TrustyGun) [inline code/game/turfs/turf.dm]: Apply 5 damage to both of their legs instead, falling on your ass should hurt like a bitch.

File: code/game/turfs/turf.dm [hunk 5]
```diff
@@ -241,12 +240,62 @@
 
 		C << "<span class='notice'>You slipped[ O ? " on the [O.name]" : ""]!</span>"
 
+		if(C.dna)
+			if(C.dna.features["tail_human"] == "Cat" && C.dna.features["ears"] == "Cat")
```
Comment (Supermichael777) [inline code/game/turfs/turf.dm]: the buff they deserve

File: code/game/turfs/turf.dm [hunk 6]
```diff
@@ -241,12 +240,62 @@
...
+				C.SpinAnimation(7,1)
+				visible_message("<span class='notice'>[C.name] slips... but manages to stick the landing with their cat like reflexs!</span>")
+				return 0
+		var/list/impact_point = pick("Head", "Chest", "Back", "Ass", "Hands", "Groin")
```
Comment (KazeEspada) [inline code/game/turfs/turf.dm]: Pickweight please

File: code/game/turfs/turf.dm [hunk 7]
```diff
@@ -241,12 +240,62 @@
 
 		C << "<span class='notice'>You slipped[ O ? " on the [O.name]" : ""]!</span>"
 
+		if(C.dna)
+			if(C.dna.features["tail_human"] == "Cat" && C.dna.features["ears"] == "Cat")
+				C.SpinAnimation(7,1)
+				visible_message("<span class='notice'>[C.name] slips... but manages to stick the landing with their cat like reflexs!</span>")
```
Comment (Cheridan) [inline code/game/turfs/turf.dm]: reflexs!

File: code/game/turfs/turf.dm [hunk 8]
```diff
@@ -241,12 +240,62 @@
...
+					if(H.armor["melee"])
+						playsound(C, 'sound/effects/woodhit.ogg', 30, 1)
+						message = "[C.name] slips backwards straight onto their head! You hear a crack as their [H.name] takes the impact!"
+						H.armor["melee"] = 0
+						H.desc += " A nasty looking crack runs right up the middle, this won't be good for stopping impacts anymore."
+				if(!message) //Should have worn your helmet!
+					if(prob(25))
+						playsound(C, 'sound/effects/blobattack.ogg', 30, 1)
+						message = "[C.name] slips backwards straight onto their head! You hear a sickening crack as their skull splits open!"
+						C.apply_damage(75, BRUTE, "head")
```
Comment (Iamgoofball) [inline code/game/turfs/turf.dm]: :-1: use adjustBruteLoss

File: code/game/turfs/turf.dm [hunk 9]
```diff
@@ -241,12 +240,62 @@
...
+
+			if("Hands")
+				message = "[C.name] slips forwards, holding out their hands to catch their fall!"
+				C.accident(C.l_hand)
+				C.accident(C.r_hand)
+
+			if("Groin")
+				message = "[C.name] slips into a split, smacking into the ground in the most unfortunate way possible!"
+				if(C.gender == MALE)
+					playsound(C, 'sound/items/bikehorn.ogg', 30, 1)
```
Comment (Iamgoofball) [inline code/game/turfs/turf.dm]: needs to change the gender

File: code/game/turfs/turf.dm (hunk 6 above)
Comment (Iamgoofball) [inline code/game/turfs/turf.dm]: no

File: code/game/turfs/turf.dm (hunk 8 above)
Comment (Shapsy) [inline code/game/turfs/turf.dm]: Can that do zone damage though?

Comment (Incoming5643) [inline code/game/turfs/turf.dm]: Why ya people gotta be pedantic about one day meme pulls.

**Comment by duncathan** [comment]
this is a huge buff to catgirls

**Comment by Incoming5643** [comment]
This is as complete as it's gonna get. It's already april fools day in china

File: code/game/turfs/turf.dm (hunk 8 above)
Comment (ChangelingRain) [inline code/game/turfs/turf.dm]: it can't, do, this is the correct way to do it

**Pull request merged**